analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TooManyParameters_CustomValues.CSharp11.cs

```diff
@@ -17,4 +17,46 @@ public interface If
         static abstract void F1(int p1, int p2, int p3);
         static abstract void F2(int p1, int p2, int p3, int p4); // Noncompliant
     }
+
+    public unsafe partial class LibraryImportAttributeImports
+    {
+        [LibraryImport("foo.dll")]
+        public static partial void ExternWithoutMarshaling(int p1, int p2, int p3, int p4); // Compliant, external definition
+
+        // Provide the implementation usually generated by the source generator
+        [System.Runtime.InteropServices.DllImportAttribute("foo.dll", EntryPoint = "Extern", ExactSpelling = true)]
+        public static extern partial void ExternWithoutMarshaling(int p1, int p2, int p3, int p4);
```

Comment:
is this necessary, for this test case? or it won't compile otherwise?

Reply:
It is a copy of the original generated code by the SG. It is needed for two reasons:
* It doesn't compile without it (the partial needs an implementation)
* The test would be incomplete because it would miss the case where the SG modifies the partial definition:
  * Add `[DllImportAttribute]`
  * Add `extern`
I want to keep it unmodified with all the probably unneeded cruft, like the entire namespace and additional attribute arguments, to make it clear it was copied from the original and can be actualized by copying it again if things change.

Reply:
What are the instructions to regenerate this? 
Maybe add an Markdown documentation file in our docs for this? (as it's a generic topic)